### PR TITLE
inject buffer for the thunk bundle

### PR DESCRIPTION
**Pull request by huayang-codaio**
Per feedback from https://github.com/coda/packs-sdk/pull/1274, we don't want to browserify the entire thunk bundle. For now, the only node module used by thunk is the Buffer object. 

This PR injects Buffer with a shim and avoided browserify'ing the whole bundle. 

thunk_bundle.js size increase from 106K to 170K.

### CODE REVIEW

**Comment by huayang-codaio** [comment]
PTAL @chris-codaio @jonathan-codaio

**Review by jonathan-codaio — APPROVED** [review]
This seems reasonable to me.

File: dist/runtime/common/marshaling/marshal_buffer.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ const constants_1 = require("./constants");
 const constants_2 = require("./constants");
 function marshalBuffer(val) {
     // Buffer is not provided by IVM. If the bundle is not browserified, global.Buffer will be undefined.
-    if (global.Buffer && val instanceof Buffer) {
+    if (val instanceof Buffer) {
```
Comment (chris-codaio) [inline dist/runtime/common/marshaling/marshal_buffer.js:8]: Is the comment above here still valid?

File: Makefile [hunk 2]
```diff
@@ -66,6 +66,7 @@ compile:
 	${ROOTDIR}/node_modules/.bin/esbuild ${ROOTDIR}/runtime/thunk/thunk.ts \
 		--bundle \
 		--outfile=${ROOTDIR}/bundles/thunk_bundle.js \
+		--inject:${ROOTDIR}/testing/injections/buffer_shim.js \
```
Comment (chris-codaio) [inline Makefile:69]: Why are you injecting code from a `testing` directory into the bundle?

**Pull request merged**